### PR TITLE
demo: Scrolling issue demo [do not merge]

### LINKED ISSUE #2045: Vertical scrollbar is shown inside the horizontally scrollable area, which makes it usually out of sight
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and [Parse Dashboard](https://github.com/parse-community/parse-dashboard/releases). 

### Issue Description

To see the vertical scrollbar, one has to scroll to the far right of the table, not on the screen, because many tables have many columns, and by going there, one no longer sees whatever horizontal position they were at, and in very wide tables, it may take a while to go back to the spot they were originally at, before looking at the scrollbar. Parse Dashboard is the only app I know that does this (among the ones kind enough to display a scrollbar).

### Steps to reproduce

Open any table where columns take more than one screen wide, in any version of Parse Dashboard.

### Actual Outcome

Vertical scrollbar only appears when horizontal scrollbar at maximum position (right end).

### Expected Outcome

Vertical scrollbar always visible.

### Environment

Dashboard
- Parse Dashboard version: both 2.1.0 release and in 4.0.0 release.
- Browser (Safari, Chrome, Firefox, Edge, etc.): Firefox
- Browser version: 97 and several previous ones

Server
- Parse Server version: 4.3.0
- Operating system: Heroku 18
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): Heroku

Database
- System (MongoDB or Postgres): Mongo
- Database version: ?
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): Heroku

### PR DESCRIPTION
**Pull request by mtrezza**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

### Issue Description

Closes: https://github.com/parse-community/parse-dashboard/issues/2045#issuecomment-1609537846

### Approach

This is just a demo for the scrolling issue that has been introduced with https://github.com/parse-community/parse-dashboard/pull/2457. See https://github.com/parse-community/parse-dashboard/issues/2045#issuecomment-1609537846.

### CODE REVIEW

**Comment by mtrezza** [comment]
@AshishBarvaliya here are the instructions to replicate the scroll issue described in https://github.com/parse-community/parse-dashboard/issues/2045#issuecomment-1609537846:

1. Open the [preview environment](https://github.com/parse-community/parse-dashboard/pull/2486#issuecomment-1611988275)
2. Open class "MyClass"
3. Scroll to the very right and very bottom corner
4. Open class "_User"
5. Notice that the scroll position is already strange, there is a lot of blank space to scroll around, even thought there are 0 rows
6. Click on "Add row" button in the top bar
7. Scroll to the top (but keep scrolled to the very right)
8. Scroll to the left and see how the headers move independently from the row fields


https://github.com/parse-community/parse-dashboard/assets/5673677/06a0ebf1-eec4-421a-977e-95f0b9d04d88

**Comment by AshishBarvaliya** [comment]
@mtrezza  I don't know what I am missing here but I am still not able to reproduce the issue.

ezgif-1-17d30cae2c

**Comment by mtrezza** [comment]
Looks like a browser compatibility issue. I'm using Safari Version 16.4 (18615.1.26.110.1) and I'm able to reproduce the issue with the preview environment and following the same steps in your video. What browser are you using?

**Comment by mtrezza** [comment]
@AshishBarvaliya Were you able to reproduce this?

File: src/dashboard/Data/Browser/Browser.scss [hunk 1]
```diff
@@ -14,12 +14,13 @@
...
   bottom: 0;
   overflow: auto;
-  padding-top: 30px;
+  width: calc(100% - 300px);
 }
 
 body:global(.expanded) {
   .browser {
     left: $sidebarCollapsedWidth;
+    width: calc(100% - $sidebarCollapsedWidth);
```
Comment (AshishBarvaliya) [inline src/dashboard/Data/Browser/Browser.scss]: Remove: `width: calc(100% - $sidebarCollapsedWidth);`

File: src/dashboard/Data/Browser/Browser.scss [hunk 2]
```diff
@@ -14,12 +14,13 @@
   right: 0;
   bottom: 0;
   overflow: auto;
-  padding-top: 30px;
+  width: calc(100% - 300px);
```
Comment (AshishBarvaliya) [inline src/dashboard/Data/Browser/Browser.scss]: Remove: `width: calc(100% - 300px);` / `padding-top: 30px;`
Add: `display: flex;`

File: src/components/DataBrowserHeaderBar/DataBrowserHeaderBar.scss [hunk 3]
```diff
@@ -8,7 +8,7 @@
 @import 'stylesheets/globals.scss';
 
 .bar {
-  position: absolute;
+  position: sticky;
   top: 0;
   left: 0;
   height: 30px;
```
Comment (AshishBarvaliya) [inline src/components/DataBrowserHeaderBar/DataBrowserHeaderBar.scss:14]: Line no. 17, 18 Replace with - `display: flex; flex: 1;`

**Comment by AshishBarvaliya** [comment]
Seems like `width: calc(100% - 300px);` was causing the issue in Safari.  Please try with this alternative suggestions.

File: src/dashboard/Data/Browser/Browser.scss (hunk 2 above)
Comment (mtrezza) [reply src/dashboard/Data/Browser/Browser.scss:17]:
```suggestion
  display: flex;
```

File: src/dashboard/Data/Browser/Browser.scss (hunk 1 above)
Comment (mtrezza) [reply src/dashboard/Data/Browser/Browser.scss:23]:
```suggestion
```

**Comment by mtrezza** [comment]
Done, lets see how it goes when the preview env is recreated

**Comment by mtrezza** [comment]
The issue now is that when scrolling into the bottom right corner, and then changing to a different class with many rows, the data browser does not start at the top most (first) row, but at a lower row.

https://github.com/parse-community/parse-dashboard/assets/5673677/350478f2-f27f-4311-bebc-9c47538b7376

**Pull request closed without merging**